Index: OS/src/OSCommonInterfaces/OSInstance.cpp
===================================================================
--- OS/src/OSCommonInterfaces/OSInstance.cpp	(revision 2645)
+++ OS/src/OSCommonInterfaces/OSInstance.cpp	(working copy)
@@ -4698,3 +4698,12 @@
 }//end getADSparsityHessian()
 
 
+void OSInstance::notifyVariableChange() {
+       if(m_bProcessVariables) {
+               delete[] m_mcVariableTypes;
+               delete[] m_msVariableNames;
+               delete[] m_mdVariableLowerBounds;
+               delete[] m_mdVariableUpperBounds;
+               m_bProcessVariables = false;
+       }
+}
Index: OS/src/OSCommonInterfaces/OSInstance.h
===================================================================
--- OS/src/OSCommonInterfaces/OSInstance.h	(revision 2645)
+++ OS/src/OSCommonInterfaces/OSInstance.h	(working copy)
@@ -740,6 +740,7 @@
  */
 class OSInstance {
 public:
+        void notifyVariableChange();
 
 	/** The OSInstance class constructor */
 	OSInstance(); 
