Index: OS/src/OSCommonInterfaces/OSInstance.cpp
===================================================================
--- OS/src/OSCommonInterfaces/OSInstance.cpp	(Revision 2532)
+++ OS/src/OSCommonInterfaces/OSInstance.cpp	(Arbeitskopie)
@@ -4313,3 +4313,12 @@
 }//end getADSparsityHessian()
 
 
+void OSInstance::notifyVariableChange() {
+       if(m_bProcessVariables) {
+               delete[] m_mcVariableTypes;
+               delete[] m_msVariableNames;
+               delete[] m_mdVariableLowerBounds;
+               delete[] m_mdVariableUpperBounds;
+               m_bProcessVariables = false;
+       }
+}
Index: OS/src/OSCommonInterfaces/OSInstance.h
===================================================================
--- OS/src/OSCommonInterfaces/OSInstance.h	(Revision 2532)
+++ OS/src/OSCommonInterfaces/OSInstance.h	(Arbeitskopie)
@@ -1374,7 +1374,9 @@
 	
 public:
 
-	/**
+	void notifyVariableChange();
+
+		/**
 	 * Get instance name. 
 	 * @return instance name. Null or empty std::string ("") if there is no instance name. 
 	 */
